src/DemoAPI/API.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stock2Shop\Connector\DemoAPI;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Stock2Shop\Share;
+
+class API
+{
+    private readonly string $url;
+
+    public function __construct(string $url)
+    {
+        $this->url = $url;
+    }
+
+    /**
+     * @return Product[]
+     * @throws GuzzleException
+     */
+    public function getProducts(string $fromID, int $limit): array
+    {
+        // create guzzle client
+        $client = new Client([
```

Comment:
guzzle client set in constructor rather?
The methods are not static so he constructor will always be called/